lib/src/screens/exchange/exchange_page.dart

```diff
@@ -501,6 +381,111 @@ class ExchangePage extends BasePage {
         ));
   }
 
+  Widget _buildTemplateSection(BuildContext context) {
+    return Container(
+      height: 40,
+      width: double.infinity,
+      padding: EdgeInsets.only(left: 24),
+      child: SingleChildScrollView(
+        scrollDirection: Axis.horizontal,
+        child: Observer(
+          builder: (_) {
+            final templates = exchangeViewModel.templates;
+            final itemCount = templates.length;
+            return Row(
+              children: <Widget>[
+                GestureDetector(
+                  onTap: () =>
+                      Navigator.of(context).pushNamed(Routes.exchangeTemplate),
+                  child: Container(
+                    padding: EdgeInsets.only(left: 1, right: 10),
+                    child: DottedBorder(
+                      borderType: BorderType.RRect,
+                      dashPattern: [6, 4],
+                      color: Theme.of(context)
+                          .primaryTextTheme
+                          .display2
+                          .decorationColor,
+                      strokeWidth: 2,
+                      radius: Radius.circular(20),
+                      child: Container(
+                        height: 34,
+                        padding: EdgeInsets.only(left: 10, right: 10),
+                        alignment: Alignment.center,
+                        decoration: BoxDecoration(
+                          borderRadius: BorderRadius.all(Radius.circular(20)),
+                          color: Colors.transparent,
+                        ),
+                        child: templates.length >= 1
+                            ? Icon(
+                                Icons.add,
+                                color: Theme.of(context)
+                                    .primaryTextTheme
+                                    .display3
+                                    .color,
+                              )
+                            : Text(
+                                S.of(context).new_template,
+                                style: TextStyle(
+                                  fontSize: 14,
+                                  fontWeight: FontWeight.w600,
+                                  color: Theme.of(context)
+                                      .primaryTextTheme
+                                      .display3
+                                      .color,
+                                ),
+                              ),
+                      ),
+                    ),
+                  ),
+                ),
+                ListView.builder(
+                  scrollDirection: Axis.horizontal,
+                  shrinkWrap: true,
+                  physics: NeverScrollableScrollPhysics(),
+                  itemCount: itemCount,
```

Comment:
if you use `itemCount` only in this place then please replace by `templates.length` and remove `itemCount` variable